src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1094,14 +1096,14 @@ private FunctionGroup GetFuncGroup(RuntimeCore runtimeCore, List<StackValue> arg
             }
 
             // use static classScope
-            return FirstFunctionGroupInInheritanceChain(runtimeCore, classScope);
+            return FirstFunctionGroupInInheritanceChain(methodName, classScope, classTable, globalFunctionTable);
         }
 
-        private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCore, int cidx)
+        internal static FunctionGroup FirstFunctionGroupInInheritanceChain(string methodName, int cidx, ClassTable classTable, FunctionTable globalFnTable)
         {
             FunctionGroup funcGroup = null;
-            var classNodes = runtimeCore.DSExecutable.classTable.ClassNodes;
-            var globalFuncTable = globalFunctionTable.GlobalFuncTable;
+            var classNodes = classTable.ClassNodes;
+            var globalFuncTable = globalFnTable.GlobalFuncTable;
```

Comment:
What's the real advantage of making this function `static `or is it just cosmetic?

Reply:
it is just a refactor so that it can be shared by the MSIL APIs